### DISCUSSION TITLE
sdk use Case

### DISCUSSION THREAD

**Discussion by yolk23321** [openlit/openlit#552] [Q&A] [answered]
I want to use Openlit to monitor the API calls of Gemini, but I couldn't find a complete example code. Could you provide me with one? I hope the official documentation can be supplemented with more detailed examples for the SDK

**Comment by amanagarwal042** [comment]
i can help you, are you using vertex ai or google ai studio?

**Reply by yolk23321** [reply]
google ai studio

**Reply by amanagarwal042** [reply]
Here is an example, This will print all traces and metrics directly to you console. If you need to send this to an OTel backend and store the traces and metrics for analysis, You can refer [this](https://github.com/openlit/openlit/tree/main?tab=readme-ov-file#step-3-initialize-openlit-in-your-application)

```python
import google.generativeai as genai
import openlit

# Initialize environment and application name for OpenLIT monitoring
openlit.init(environment="testing", application_name="test")

# Initialize Google AI Studio client
genai.configure(api_key=os.getenv("GOOGLE_AI_STUDIO_API_TOKEN"))
model = genai.GenerativeModel("gemini-1.5-flash")

response = model.generate_content("Observability for LLMs", stream=False)

# streaming response
response = model.generate_content("Monitor AI", stream=True)
for text in response:
    assert isinstance(text.text, str)
```

**Reply by amanagarwal042** [reply, marked as answer]
@yolk23321 lemme know if you have any issues that I can help with
